cmd/download.go

```diff
@@ -0,0 +1,284 @@
+func getChembl(writePath string) {
+	links, err := getPageLinks("https://ftp.ebi.ac.uk/pub/databases/chembl/ChEMBLdb/latest/")
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var sqliteFileLink string
+
+	// find the sqlite file link
+	for _, link := range links {
+		// if it's a sqlite tarball save its link
+		if strings.Contains(link, "sqlite.tar.gz") {
+			sqliteFileLink = link
+			break
+		}
+	}
+
+	// if we didn't find it, bail.
+	if sqliteFileLink == "" {
+		log.Fatal("could not find sqlite file link")
+	}
+
+	// get the tarball from the server that contains the sqlite file
+	response, err := http.Get(sqliteFileLink)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer response.Body.Close()
+
+	// if server ain't good, bail
+	if response.StatusCode != 200 {
+		log.Fatalf("status code error: %d %s", response.StatusCode, response.Status)
+	}
+
+	// extract our sqlite file from the tarball and write to disk
+	err = getTarballFile(response.Body, ".db", writePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// getGenbank checks the latest release of Genbank, grabs all files ending with .gz extension. Decompresses them and saves to disk location specified by writePath.
+func getGenbank(writePath string) {
+	writePathDirectory := filepath.Join(writePath, "genbank")
+	links, err := getPageLinks("https://ftp.ncbi.nlm.nih.gov/genbank")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, link := range links {
+		parsedURL, err := url.Parse(link)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		filename := filepath.Base(parsedURL.Path)
+		extension := filepath.Ext(filename)
+
+		if extension == ".gz" { // if it's a gzipped file it's a genbank file so download and unzip it
+			fmt.Println("retrieving: " + link)
+			go getFile(link, writePathDirectory)
+		}
+	}
+}
+
+// getFile downloads the file at the specified url and saves it to the specified writePath.
+func getFile(fileURL string, writePath string) {
+	// get the file from the server
+	response, err := http.Get(fileURL)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer response.Body.Close()
+
+	// if server ain't good, bail
+	if response.StatusCode != 200 {
+		log.Fatalf("status code error: %d %s", response.StatusCode, response.Status)
+	}
+
+	// if the filepath does not exist, create it
+	err = os.MkdirAll(writePath, os.ModePerm)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// parse url for filename
+	parsedURL, err := url.Parse(fileURL)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	filename := filepath.Base(parsedURL.Path)
+	pathname := filepath.Join(writePath, filename)
+
+	// create a new file to write the data to it
+	file, err := os.Create(pathname)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+
+	// copy the file to disk
+	if _, err := io.Copy(file, response.Body); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func getPageLinks(url string) ([]string, error) {
```

Comment:
Some more comments here for people who don't understand the basics of html parsing and http requests.